Tests/ApolloCodegenTests/ApolloCodegenConfigurationCodableTests.swift

```diff
@@ -923,4 +923,49 @@ class ApolloCodegenConfigurationCodableTests: XCTestCase {
       ))
     )
   }
+
+  func test__decodeApolloCodegenConfiguration__withInvalidOptions() throws {
```

Comment:
To prevent regressions and ensure that we are actually calling the validation function in each of the decoder methods (or if we move to a protocol, that all of the proper objects conform to and use the `StrictlyKeyedDecodable` protocol), we should probably have multiple tests here, one where there is an invalid key in each of the objects.

Reply:
Added more tests -- and an additioanl one to handle mutliple failures in a single object.